### ISSUE TITLE
Unable to get 3 and 4 controllers working

### ISSUE THREAD

**Issue by frdfsnlght** [luni64/TeensyStep#25]
I'm trying to get multiple controllers moving motors at the same time. I have no problems until I get to the 3rd controller, then the Teensy hangs. In fact, any time I try to use the 3rd controller, even synchronously, it hangs. I've attached a sample project that illustrates what I'm trying to do (had to use a txt extension for upload).

I'm using a Teensy 3.2.

[Teensy-Stepper.txt](https://github.com/luni64/TeensyStep/files/2385894/Teensy-Stepper.txt)

**Comment by luni64** [comment]
I did a quick test with a T3.2 (not connecting motors, just looking at the output) and it works nicely. Nothing hangs, Motors stop as expected some 10s after the start. 

**Please note:** I fixed a bug a couple of weeks ago related to problems with more than 2 controllers [(issue 24)](https://github.com/luni64/TeensyStep/issues/24). Are you sure that you are using the current version? 

If you are already using the current version, can you please describe the problem in more detail?

------
Few comments on your code (not related to the issue) 

![image](https://user-images.githubusercontent.com/12611497/45593329-86038780-b984-11e8-8341-76f8fabb3e1c.png)

- Not necessary to set the pin mode of the step and DIR pins, this is handled by the library. 
- You only use **one** DIR output for all motors. This will usually not work! In your example all motors move in the same direction so it doesn't harm. As soon as one motor needs to move in the opposite direction you'll get a mess. 



- There seems to be a bug in your setup(): 
![image](https://user-images.githubusercontent.com/12611497/45593305-a97a0280-b983-11e8-8313-5951ff8cb76b.png)
You probably meant to set the maxSpeed for all motors but you only set it for mot0. The others stay at their default value. 

I like your clear and nicely writen code very much! Just one thing which somehow "hurts my heart": I never understood why people stick with those unnecessary and ugly #defines. c++ has "constexpr" which is a much better, typesafe replacement for defines.
 
![image](https://user-images.githubusercontent.com/12611497/45593394-07a7e500-b986-11e8-8fe9-2690494e1e3a.png)

**Comment by frdfsnlght** [comment]
Library version was the problem. I was using 0.97 and you had changed some things in 0.98. I didn't realize you had updated. It works as expected now. Thank you very much!

And thanks for the code review!

* I took out the lines for setting pin directions for the motor as you noted.
* I'm intentionally using the same pin for direction for all the steppers. My project will only ever drive all the motors in the same direction at the same time, so no problem.
* I fixed the bug you found. Thanks.
* I'll start using "constexpr" in my projects. I wasn't aware of it. I don't write much C++ these days.

I've attached the latest code for posterity.

And a little background: I'm using your library to drive 16 steppers from a single Teensy 3.2. I only need to drive 4 independently and simultaneously except when I need to drive all 16 synchronously. Your library is perfect for this. This is part of a Barbot I'm building. Each motor drives a 3D printed pump for a single ingredient. The Teensy is but a small, but very important part of this project. There's also an Arduino Pro Mini and a Raspberry Pi with 7" touchscreen. I hope to have it finished by Christmas when I'll need it to survive all the holiday "cheer" and family functions.

[Teensy-Stepper.txt](https://github.com/luni64/TeensyStep/files/2386692/Teensy-Stepper.txt)

**Comment by luni64** [comment]
Sounds like a great project, I'd love to put a video of that barbot in the TeensyStep readme. Just drop me a note if you have something and want me to put it in an "examples section"

**Comment by conkerts** [comment]
Hi, I seem to have weird behavior with 3 and 4 controller objects as well.
I just recently cloned the repo (when I wrote my first question here), and I checked I have the version=0.98 .

Basically I've got 4 Motor objects and 4 controllers. It first works for some commands, then I get weird motor movements, even I send commands to a completely different controller & motor.
(Teensy 3.5 btw , with all current Arduino versions)

Should I post my details here, or create a new, separate issue ? 
And how can I debug this issue most effectively ? .
